ocs_ci/ocs/registry.py

```diff
@@ -343,3 +345,67 @@ def check_image_exists_in_registry(image_url):
         return_value = True
         logger.info("Image exists in Registry")
     return return_value
+
+
+def image_pull_and_push(
+    project_name, template, image='', pattern='', wait=True
+):
+    """
+    Pull and push images running oc new-app command
+
+    Args:
+        project_name (str): Name of project
+        template (str): Name of the template of the image
+        image (str): Name of the image with tag
+        pattern (str): name of the build with given pattern
+        wait (bool): If true waits till the image pull and push completes.
+
+    """
+    ocp_obj = ocp.OCP()
+    ocp_obj.new_project(project_name=project_name)
+    if config.DEPLOYMENT.get('disconnected'):
+        mirrored_image, authfile_name = mirror_image(image=image)
+        cmd = f"podman pull {mirrored_image} --authfile {authfile_name}"
+        ocp_obj.exec_oc_cmd(command=cmd, out_yaml_format=False)
+    else:
+        cmd = f"new-app --template={template} -n {project_name}"
+        ocp_obj.exec_oc_cmd(command=cmd, out_yaml_format=False)
+
+        # Validate it completed
+        if wait:
+            wait_time = 300
```

Comment:
this var is not in use

Reply:
Thanks for pointing it out, added now

Reply:
NP. 
Is there a way to poll the build status instead of sleep?

Reply:
It takes some time to bring the build up(the name of the build will get once it is up and start processing). In below lines we are checking and waiting till build status reach `Complete`.